azure/keyvault/main.tf

```diff
@@ -2,11 +2,14 @@
 
 # Create the Azure Key Vault
 resource "azurerm_key_vault" "key_vault" {
-  name                = var.name
-  location            = var.location
-  resource_group_name = var.resource_group_name
-  tenant_id           = data.azurerm_client_config.current.tenant_id
-  sku_name            = "standard"
+  name                       = var.name
+  location                   = var.location
+  resource_group_name        = var.resource_group_name
+  tenant_id                  = data.azurerm_client_config.current.tenant_id
+  sku_name                   = "standard"
+  soft_delete_retention_days = 7
+
+  purge_protection_enabled = var.isProtect
```

Comment:
Purge is optional.

Reply:
To ignore warnings, it might be a good idea to add a special comment at the top of your code as shown in `tfsec` [documentation](https://aquasecurity.github.io/tfsec/v1.8.0/getting-started/configuration/ignores/).

In this case, the rule ID is named `#tfsec:ignore:azure-keyvault-no-purge`, and you could add a comment on the top of your code that should look like this:

```hcl
  #tfsec:ignore:azure-keyvault-no-purge
  purge_protection_enabled = var.isProtect
```
or to the side:

```hcl  
  purge_protection_enabled = var.isProtect #tfsec:ignore:azure-keyvault-no-purge
```

>Note: if you want to add an expiration date, you can do it by adding a postfix `:exp:<YYYY-MM-DD>` to the comment. In this case, for example, you could do something like that: `#tfsec:ignore:azure-keyvault-no-purge:exp:2023-01-01`